File: music-module/src/main/java/per/zsck/music/entity/Music.java
package per.zsck.music.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import per.zsck.music.utils.MusicUtil;

/**
 * @author zsck
 * @date 2022/12/11 - 11:15
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Document("music")
public class Music {

    @Id
    String id;
    String audioName;
    /**
     * 根据audioName设置
     */
    String audioNameIndex;
    String title;
    String artist;
    String md5;
    /**
     * 根据fileId设置
     */
    String url;
    String fileName;

    ObjectId fileId;

    public Music(String audioName, String md5, ObjectId fileId) {
        this.setAudioName( audioName );
        this.md5 = md5;
        this.setFileId(fileId );
    }

    public void setFileId(ObjectId fileId) {
        this.fileId = fileId;
        this.url = MusicUtil.URL_PREFIX + fileId.toString();
    }

    public void setAudioName(String audioName) {
        this.audioName = audioName;
        this.audioNameIndex = audioName.replaceAll("\\s", "");
    }
}

File: src/main/kotlin/per/zsck/simbot/http/kugou/entity/Music.kt
package per.zsck.simbot.http.kugou.entity

import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import org.springframework.data.annotation.Transient
import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

/**
 * @author zsck
 * @date   2022/11/10 - 12:37
 */
@Document("music")
class Music{

    @Id
    var id: String? = null

    @JsonAlias("FileName")
    var audioName: String? = null
        set(value) {
            audioNameIndex = value?.replace("\\s", "")
            field = value
        }

    /**
     * 不包含空格的audioName，用于模糊查询
     */
    var audioNameIndex: String? = null

    @JsonAlias("SongName", "song_name")
    var title: String? = null

    @JsonAlias("author_name")
    var artist: String? = null
    var md5: String? = null

    @JsonAlias("play_backup_url")
    var url: String? = null

    @Transient
    @JsonAlias("img")
    var imgUrl: String? = null

    @JsonAlias("FileHash")
    @Transient
    var fileHash: String? = null

    @Transient
    @JsonAlias("AlbumID")
    var albumID: String? = null

    var file: ByteArray? = null
}